### PR TITLE
📖 Add release guidelines and policies to contributing guide

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
/assign @ncdc @timothysc @fabriziopandini @neolit123 @CecileRobertMichon @detiber

### CODE REVIEW

File: CONTRIBUTING.md
Comment (ncdc) [inline CONTRIBUTING.md:67]: I think we might need more details here
- Only critical bug fixes may be merged in between freeze & release
- Have to be approved by at least n (2?) project maintainers
- ... ?

Comment (vincepri) [reply CONTRIBUTING.md]: done ptal

File: CONTRIBUTING.md
Comment (sedefsavas) [inline CONTRIBUTING.md:69]: Currently only 1  `approve` is enough. Is this going to be enforced somewhere?

Comment (vincepri) [reply CONTRIBUTING.md:69]: Probably not, just going to be diligent about it with a `/hold`

File: CONTRIBUTING.md
Comment (CecileRobertMichon) [inline CONTRIBUTING.md:66]: How will the code freeze be communicated and enforced? Will the main branch still be open for merges during that time?

File: CONTRIBUTING.md
Comment (fabriziopandini) [reply CONTRIBUTING.md:69]: might be add something like
```
In order to achieve this, during code-freeze:
- the first approver should apply /approve followed by /hold
- the second approver should apply /approve followed by /hold cancel
```

**Review by fabriziopandini — COMMENTED** [review]
/approve

**Comment by fabriziopandini** [comment]
/hold
due to code freeze 😅

**Comment by fabriziopandini** [comment]
/approve cancel
/lgtm 
lgtm before approval according to https://github.com/kubernetes-sigs/cluster-api/blob/master/CONTRIBUTING.md#approvals

File: CONTRIBUTING.md
Comment (vincepri) [reply CONTRIBUTING.md:66]: Probably depends, if a release branch is created we can merge PRs in, otherwise we should probably wait

In general, I don't think we need to hold every single PR (that would create lots of noise), we can leave PRs without explicit approvals until the very last minute

Comment (vincepri) [reply CONTRIBUTING.md:66]: In terms of communication, we should communicate at least 1 week before the release at one of the community meetings, I'll make it explicit in here

File: CONTRIBUTING.md
Comment (wfernandes) [inline CONTRIBUTING.md:64]: nit: I know this might be obvious but I'm assuming that `year` is calendar year. For example, Jan 1, 2020 - Jan 1, 2021.

Comment (vincepri) [reply CONTRIBUTING.md:64]: Definitely can add it if you prefer, although not sure if that's a hard rule that it was to be in a calendar year

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm

**Pull request merged**
